api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+     * annotation appears on the type declaration, or any of it's members it will be removed.
+     *
+     * @param annotation the annotation to remove
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromAll(Annotation annotation);
+
+    /**
+     * Reads in from an existing AnnotatedType. Any elements not present are
+     * added. The javaClass will be read in. If the annotation already exists on
+     * that element in the configurator the read annotation will be used.
+     *
+     * @param type the type to read from
+     * @throws IllegalArgumentException if type is null
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(AnnotatedType<U> type);
+
+    /**
+     * Reads in from an existing AnnotatedType. Any elements not present are
+     * added. The javaClass will be read in if overwrite is true. If the
+     * annotation already exists on that element in the configurator, overwrite
+     * determines whether the original or read annotation will be used.
+     *
+     * @param type      the type to read from
+     * @param overwrite if true, the read annotation will replace any existing
+     *                  annotation
+     * @throws IllegalArgumentException if type is null
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(AnnotatedType<U> type, boolean overwrite);
+
+    /**
+     * Reads the annotations from an existing java type. Annotations already
+     * present will be overwritten
+     *
+     * @param type the type to read from
+     * @throws IllegalArgumentException if type is null
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(Class<U> type);
+
+    /**
+     * Reads the annotations from an existing java type. If overwrite is true
+     * then existing annotations will be overwritten
+     *
+     * @param type      the type to read from
+     * @param overwrite if true, the read annotation will replace any existing
+     *                  annotation
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(Class<U> type, boolean overwrite);
+
+    /**
+     * Override the declared type of a field
+     *
+     * @param field the field to override the type on
+     * @param type  the new type of the field
+     * @throws IllegalArgumentException if field or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideFieldType(Field field, Type type);
+
+    /**
+     * Override the declared type of a field
+     *
+     * @param field the field to override the type on
+     * @param type  the new type of the field
+     * @throws IllegalArgumentException if field or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideFieldType(AnnotatedField<? super T> field, Type type);
```

Comment:
I am not sure about usefulness of this method but if we have this then it will make sense to include something as `overrideMethodReturnType` or not? I know it's bit corner case but e.g I have `Foo` field and I overrided it to `Bar` and what about my getter method?